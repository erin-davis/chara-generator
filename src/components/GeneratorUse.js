import React from "react";
import {BrowserRouter as Router, Link, Switch, useHistory} from "react-router-dom";
import UseCard from "./UseCard.js";
//url is /use

const GeneratorUse = () =>{
  const history = useHistory();

  const sectionSelect = e =>{
    e.preventDefault();
  }
  //i had the create-card sections wrapped in Link from rrd
  const toFinal = () =>{
    history.push('/final');
  }

  const toSelect = () =>{
    history.push('/selection');
  }

//new goal, in css, display none the p but then when its hovered, display inline block

  return (
    <div className="generator-use">
        <header>
          <h1>What do you want to create?</h1>
        </header>
      <div className="creation-use">
          <section className="NPC" onClick={toFinal}>
            <h2>Non-Player Character</h2>
            <p className="NPC-gen-desc">This will bring you to a fully randomly generated character!</p>
          </section>
          <section className="PC" onClick={toSelect}>
            <h2>Player Character</h2>
            <p className="PC-gen-desc">This will allow you to have a more indepth character creation experience! You can pick and choose data you want your character to keep.</p>
          </section>
        </div>
        <div className="pick-use">
          {/*<UseCard />*/}
        </div>
    </div>
  );
}

export default GeneratorUse;
