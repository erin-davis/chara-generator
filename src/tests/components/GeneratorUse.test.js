import React from "react";
import {render} from "@testing-library/react";
import {ReactTestUtils as RTU} from "react-dom/test-utils";
import ShallowRenderer from 'react-test-renderer/shallow';
import GeneratorUse from "../../components/GeneratorUse.js";
/*a test to make sure that description in /use aka GeneratorUse.js shows only when active aka when the mouse is on it and not when the moust is not on it...*/

test('Should render /use correctly', ()=>{
  const renderer = new ShallowRenderer();
  renderer.render(<GeneratorUse />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
})

test('There is no class to start with', ()=>{
  render(
    <div>
      <GeneratorUse />
    </div>
  )

  const npcClass = document.querySelector('.NPC');
  const pcClass = document.querySelector('.PC');

  expect(pcClass.className).toBe('PC');
  expect(npcClass.className).toBe('NPC');
})

test('Should add class when mouse enters', ()=>{
  
})