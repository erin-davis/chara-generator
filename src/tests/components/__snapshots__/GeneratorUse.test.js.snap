// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render /use correctly 1`] = `
<div
  className="generator-use"
>
  <header>
    <h1>
      What do you want to create?
    </h1>
  </header>
  <div
    className="creation-use"
  >
    <section
      className="NPC"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <h2>
        Non-Player Character
      </h2>
      <p>
        This will allow you to immediately generate a character
      </p>
    </section>
    <section
      className="PC"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <h2>
        Player Character
      </h2>
      <p>
        This will allow you to have a more indepth character creation experience! You can pick and choose data you want your character to keep.
      </p>
    </section>
  </div>
  <div
    className="pick-use"
  />
</div>
`;
